lib/game_loop.rb

```diff
@@ -1,38 +1,60 @@
+require_relative './game_state'
 require_relative './guess'
-require_relative './input'
-require_relative './dialog'
 
 class GameLoop
 
-	def initialize(game_state, input = Input.new)
+	def initialize(ui, gs = GameState, g = Guess)
 		@valid = true
-		@game_state = game_state
-		@dialog = Dialog.new(game_state)
-		@input = input
+		@ui = ui
+		@gs = gs		##RENAME state: GAME STATE CLASS, G: GUESS CLASS
+		@g = g
 	end
 
 	def play
+		@game_state = @gs.new
```

Comment:
it might make more sense to just take an instance of `GameState` in your initialize method instead instead of the class itself since you're only using `@gs` to create a new game state